kbs/src/api_server.rs

```diff
@@ -0,0 +1,317 @@
+                let api_server = self.clone();
+                App::new()
+                    .wrap(middleware::Logger::default())
+                    .app_data(web::Data::new(api_server))
+                    .service(
+                        web::resource([kbs_path!("{plugin}{sub_path:.*}")])
+                            .route(web::get().to(client))
+                            .route(web::post().to(client)),
+                    )
+                    .service(
+                        web::resource([kbs_path!("admin/{plugin}/{sub_path:.*}")])
+                            .route(web::get().to(admin))
+                            .route(web::post().to(admin)),
+                    )
+            }
+        });
+
+        if !http_config.insecure_http {
+            let tls_server = http_server
+                .bind_openssl(
+                    &http_config.sockets[..],
+                    crate::http::tls_config(&http_config)
+                        .map_err(|e| Error::HTTPSFailed { source: e })?,
+                )
+                .map_err(|e| Error::HTTPSFailed { source: e.into() })?;
+
+            return tls_server
+                .run()
+                .await
+                .map_err(|e| Error::ServerLaunchFailed { source: e.into() });
+        }
+
+        http_server
+            .bind(&http_config.sockets[..])
+            .map_err(|e| Error::HTTPFailed { source: e.into() })?
+            .run()
+            .await
+            .map_err(|e| Error::ServerLaunchFailed { source: e.into() })
+    }
+}
+
+/// Client APIs. /kbs/v0/XXX
+pub(crate) async fn client(
+    request: HttpRequest,
+    body: web::Bytes,
+    core: web::Data<ApiServer>,
+) -> Result<HttpResponse> {
+    let query = request.query_string();
+    let plugin_name = request
+        .match_info()
+        .get("plugin")
+        .ok_or(Error::IllegalAccessedPath {
+            path: request.path().to_string(),
+        })?;
+    let sub_path = request
+        .match_info()
+        .get("sub_path")
+        .ok_or(Error::IllegalAccessedPath {
+            path: request.path().to_string(),
+        })?;
+
+    let end_point = format!("{plugin_name}{sub_path}");
+
+    match plugin_name {
+        #[cfg(feature = "as")]
+        "auth" if request.method() == Method::POST => core
+            .attestation_service
+            .auth(&body)
+            .await
+            .map_err(From::from),
+        #[cfg(feature = "as")]
+        "attest" if request.method() == Method::POST => core
+            .attestation_service
+            .attest(&body, request)
+            .await
+            .map_err(From::from),
+        #[cfg(feature = "as")]
+        "attestation-policy" if request.method() == Method::POST => {
+            core.admin_auth.validate_auth(&request)?;
+
+            core.attestation_service.set_policy(&body).await?;
+
+            Ok(HttpResponse::Ok().finish())
+        }
+        "resource-policy" if request.method() == Method::POST => {
```

Comment:
I think get can also implement `GET` logic here at some point to return the policy that is currently enabled.

Reply:
Yes. I will add this to `admin` api